### PR TITLE
#159961949 Sign Up Feature

### PR DESCRIPTION
**Pull request by asamisellah**
#### What does this PR do?
Allows users to create an account with authors' haven.

#### Description of Task to be completed?
- Create sign up components
- Develop actions for the sign-up component
- Develop reducer for the sign-up component
- Make input field dynamic to guide user on criteria expected
- Write tests for sign up feature

##### Prerequisites 
Node and NPM should be installed.

##### Steps

1. Clone this repo and branch `git clone git@github.com:andela/ah-jarvis-frontend.git  -b ft-signup-159961949`
2. Open the project in your prefered text editor
3. Run the code below in your terminal

```
yarn install

yarn start

```
4. You should be redirected to your browser
5. Navigate to this url: http://localhost:3000/signup
6. The sign up page should appear as below
screen shot 2018-09-07 at 11 17 45 am

#### Running tests
To run the tests use the following command in your terminal:
``` yarn test --coverage```

### Screenshots
screen shot 2018-09-11 at 3 10 03 pm

screen shot 2018-09-11 at 3 10 46 pm

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#159961949](https://www.pivotaltracker.com/story/show/159961949/comments/193936568)

### CODE REVIEW

**Review by yeenab — CHANGES_REQUESTED** [review]
Checks are failing

**Comment by asamisellah** [comment]
@j33n  I'm fixing the problems with the linter

File: src/components/SignUpForm/index.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    this.props.onClick({ user: this.state.user });
+  };
+
+  render() {
+    return (
+      <div>
+        <p className="p-t--5 red-text">
+          {this.props.register.failure
+            ? this.props.register.errors.errors.username
+            : ""}
```
Comment (yeenab) [inline src/components/SignUpForm/index.js]: Let's de-structure here, also, you can do something like this to keep your code clean:

```
const { failure, errors } = this.props.register;

{failure && errors.errors.username}
```

File: src/components/SignUpForm/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        </p>
+        <p className="p-t--10 red-text">
+          {this.props.register.failure
+            ? this.props.register.errors.errors.email
+            : ""}
+        </p>
+        <p className="p-t--10 p-b--10 red-text">
```
Comment (yeenab) [inline src/components/SignUpForm/index.js]: This paragraph looks duplicated, can we make it one component

File: src/components/SignUpForm/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                onClick={this.handleClick}
+              >
+                Sign up
+              </button>
+            </div>
+          </div>
+          <div className="row">
+            <div className="input-field col s12">
+              Already have an account?
+              <a href="https://reactjs.org/docs/thinking-in-react.html">
```
Comment (yeenab) [inline src/components/SignUpForm/index.js]: Why are we going to a `Thinking in React link`?

File: src/containers/SignUp/__test__/actions.test.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import register from "../reducer";
+import * as actions from "../actions";
+
+describe("SignUp actions", () => {
+  it("should return action type and payload for signUpSuccess", () => {
+    const user = {
+      user: {
```
Comment (yeenab) [inline src/containers/SignUp/__test__/actions.test.js]: Can we create a file that contains our `test_data` and call it in all our tests

File: src/containers/SignUp/__test__/index.test.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// import React from "react";
+// // import { SignUp } from "../index";
+// import { shallow } from "enzyme";
+// import SignUp from "..";
+
+// describe("<SignUp/>", () => {
+//   it("renders Sign Up container successfully", () => {
+//     const wrapper = shallow(<SignUp />);
+//     expect(wrapper.find("<SignUpForm/>").length).toEqual(1);
```
Comment (yeenab) [inline src/containers/SignUp/__test__/index.test.js]: Is this commented code 😢

**Review by yeenab — CHANGES_REQUESTED** [review]
### Nitpicks

File: src/components/SignUpForm/index.js (hunk 3 above)
Comment (asamisellah) [reply src/components/SignUpForm/index.js]: At the time I didn't have the sign in page link but I will change that

File: src/containers/SignUp/__test__/actions.test.js (hunk 4 above)
Comment (asamisellah) [reply src/containers/SignUp/__test__/actions.test.js]: yes, that's possible

File: src/containers/SignUp/__test__/index.test.js (hunk 5 above)
Comment (asamisellah) [reply src/containers/SignUp/__test__/index.test.js]: I've actually made a couple of changes to this feature. I will get rid of all commented code

**Comment by asamisellah** [comment]
@j33n I have made the requested changes

**Review by yeenab — APPROVED** [review]
LGTM

**Pull request merged**